patches/server/0173-Add-reason-to-potion-effect-add-event.patch

```diff
@@ -185,8 +185,35 @@ index 6d3b002b..a284d366 100644
          this.world.a((EntityHuman) null, 1017, new BlockPosition((int) this.locX, (int) this.locY, (int) this.locZ), 0);
      }
  
+diff --git a/src/main/java/net/minecraft/server/ItemFood.java b/src/main/java/net/minecraft/server/ItemFood.java
+index 4a2ec5dc..22542f26 100644
+--- a/src/main/java/net/minecraft/server/ItemFood.java
++++ b/src/main/java/net/minecraft/server/ItemFood.java
+@@ -1,5 +1,7 @@
+ package net.minecraft.server;
+ 
++import org.bukkit.event.entity.PotionEffectAddEvent.EffectAddReason;
++
+ public class ItemFood extends Item {
+ 
+     public final int a;
+@@ -35,7 +37,13 @@ public class ItemFood extends Item {
+ 
+     protected void c(ItemStack itemstack, World world, EntityHuman entityhuman) {
+         if (!world.isClientSide && this.l > 0 && world.random.nextFloat() < this.o) {
+-            entityhuman.addEffect(new MobEffect(this.l, this.m * 20, this.n));
++            // SportPaper start - add effect reason for golden apples
++            if (this instanceof ItemGoldenApple) {
++                entityhuman.addEffect(new MobEffect(this.l, this.m * 20, this.n), EffectAddReason.GOLDEN_APPLE);
++            } else {
```

Comment:
I believe other food can give effects (such as rotten flesh?). Are those handled here as well? If so they need to be handled and given the appropriate reason.